cdm/src/main/java/ucar/nc2/StringKeeper.java

```diff
@@ -0,0 +1,43 @@
+package ucar.nc2;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * A list of strings that only allows one thread to use any given value at the same time.
+ *
+ * @author cmrose
+ */
+
+public class StringKeeper {
+
+    private List stringList = Collections.synchronizedList(new ArrayList<String>());
+    private boolean waiting = false;
+
+    public synchronized void control(String item) {
```

Comment:
This will block all threads trying to control an item not just ones trying to access the item won't it? (only one thread can be running this particular bit of code at a time)  Will have the desired effect still but is overly restrictive.   Could have just added synchronized to makeUncompressed above to do this.

Reply:
My understanding is that the wait() will cause the thread to wait and to no longer be synchronised on the the object thus allowing another thread to come in and add a file to the list.  https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#wait  This is evidenced to be the case by the fact that I observe more than one file appearing in the list at any given time.

Reply:
OK, thanks for the explanation.  Can see how that works now.